cmd/gke-gcloud-auth-plugin/cred/cred.go

```diff
@@ -86,26 +58,22 @@ type cache struct {
 // plugin holds data to be passed around (eg: useApplicationDefaultCredentials)
 // as well as methods that may needs to be mocked in test scenarios.
 type plugin struct {
-	googleDefaultTokenSource         func(ctx context.Context, scope ...string) (oauth2.TokenSource, error)
-	readGcloudConfigRaw              func() ([]byte, error)
-	k8sStartingConfig                func() (*clientcmdapi.Config, error)
-	readFile                         func(filename string) ([]byte, error)
-	writeCacheFile                   func(content string) error
-	getCacheFilePath                 func() string
-	timeNow                          func() time.Time
-	useApplicationDefaultCredentials bool
+	k8sStartingConfig func() (*clientcmdapi.Config, error)
```

Comment:
Cache is strongly related to `cmd/gke-gcloud-auth-plugin/cred/gcloud_token_provider.go`, please reconsider moving those parts there.

@mikedanese, I feel that this is something in your area of expertise. Could you take a look at this PR (or delegate)?

Reply:
After this refactor, there would be a Pull Request to check in support for GDCE tokens, which will also require caching. The goal is to separate the token retrieval logic from the token caching logic, and this simplifies the onboarding for different token retrieval paths.

Reply:
SGTM